src/r_util.cpp

```diff
@@ -487,6 +488,22 @@ namespace rhost {
             return Rf_ScalarInteger(static_cast<int>(lock_state));
         }
 
+        extern "C" SEXP fetch_file(SEXP path) {
+            fs::path fpath;
+            r_top_level_exec([&]() {
+                fpath = fs::u8path(Rf_translateCharUTF8(STRING_ELT(path, 0)));
+            }, __FUNCTION__);
+
+            if (!fpath.empty()) {
+                blobs::blob file_data;
+                blobs::append_from_file(file_data, fpath);
+                host::send_notification("!FetchFile", file_data, fpath.filename().string());
```

Comment:
No, this is not invoked via eval. Scenario is: User runs some R code that writes to some file (on the remote machine). User wants to open the file locally. User executes fetch_file with the remote path.

since we don't have remote save file dialog, there is no way for us to know the path to fetch the file. so user will be triggering this.


---
In reply to: [74119967](https://github.com/Microsoft/R-Host/pull/77#discussion_r74119967) [](ancestors = 74119967)